roofit/roofitcore/src/RooAbsCollection.cxx

```diff
@@ -514,9 +524,17 @@ Bool_t RooAbsCollection::add(const RooAbsArg& var, Bool_t silent)
 /// Add a collection of arguments to this collection by calling addOwned()
 /// for each element in the source collection
 
-Bool_t RooAbsCollection::addOwned(const RooAbsCollection& list, Bool_t silent)
+bool RooAbsCollection::addOwned(const RooAbsCollection& list, bool silent)
 {
-  Bool_t result(false) ;
+  if(list.isOwning()) {
+    throw std::runtime_error("Passing an owning RooAbsCollection by const& to"
```

Comment:
Suggest `std::invalid_argument` when you get an invalid argument.

Reply:
Thanks, good suggestion! I have updated it.